File: 206-reverse-linked-list/206-reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        count = 0
        prev_node = head
        next_node = head
        while next_node is not None:
            if count == 0:
                next_node = head.next
                head.next = None
                count += 1
            else:
                head = next_node
                next_node = head.next
                head.next = prev_node
                prev_node = head
        return head
                
            

File: 206-reverse-linked-list/206-reverse-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        if(head == null) {
            return head;
        }
        ListNode prev = null;
        ListNode current = head;
        head = head.next;
        while(head != null) {
            current.next = prev;
            prev = current;
            current = head;
            head = current.next;
        }
        current.next = prev;
        return current;
    }
}